src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -175,8 +175,10 @@ private async Task<IFormCollection> InnerReadFormAsync(CancellationToken cancell
                     while (section != null)
                     {
                         // Parse the content disposition here and pass it further to avoid reparsings
-                        ContentDispositionHeaderValue contentDisposition;
-                        ContentDispositionHeaderValue.TryParse(section.ContentDisposition, out contentDisposition);
+                        if (!ContentDispositionHeaderValue.TryParse(section.ContentDisposition, out var contentDisposition))
+                        {
+                            throw new InvalidDataException("Form section has invalid Content-Disposition value.");
```

Comment:
nit: It should show the raw value in the exception message

Reply:
It would mean that untrusted (user provided) data is reflected in error message. It is not good security practice. Do you really want to do that here?

Reply:
Yes, it should be in there. Exceptions don't make it back to the client at all in a proper application as they themselves have more revealing information (like the stack trace, line numbers etc).

Reply:
I added change since there are already many such exceptions with raw data. 
However, I think it would have been nice to keep InvalidDataException messages safe (defense in depth). 
I would have preferred  to return generic user friendly exception message and having raw value logged.